decrypt_oracle/src/aws_encryption_sdk_decrypt_oracle/key_providers/counting.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""
+Master key that generates deterministic data keys and decrypts a pre-defined
+encrypted data key value to that deterministic data keys.
+"""
+from typing import Dict, NoReturn, Text
+
+from aws_encryption_sdk.exceptions import DecryptKeyError
+from aws_encryption_sdk.identifiers import AlgorithmSuite
+from aws_encryption_sdk.key_providers.base import MasterKey, MasterKeyConfig
+from aws_encryption_sdk.structures import EncryptedDataKey
+from aws_encryption_sdk.structures import DataKey
+
+
+class CountingMasterKeyConfig(MasterKeyConfig):
+    # pylint: disable=too-few-public-methods
+    """Passthrough master key configuration to set the key id to "test_counting_prov_info"."""
+
+    provider_id = "test_counting"
+
+    def __init__(self) -> None:
+        """Set the key id to "test_counting_prov_info"."""
+        super(CountingMasterKeyConfig, self).__init__(key_id=b"test_counting_prov_info")
+
+
+class CountingMasterKey(MasterKey):
+    """Master key that generates deterministic data keys and decrypts a pre-defined
```

Comment:
Interesting piece of test code. Are we using it elsewhere?

It should come with a massive warning that it is *not* secure and must not be used for any non-test cases.

Reply:
Not to my knowledge, no. Warning added in docstring that this should never be used outside of testing.